poc/extensions/src/extension_fungibles.rs

```diff
@@ -0,0 +1,42 @@
+use crate::{DispatchError, Dispatchable};
+use crate::{ExtensionId, ExtensionIdTy};
+use core::marker::PhantomData;
+use parity_scale_codec::{Decode, Encode};
+
+pub trait ExtensionFungibles: ExtensionId {
+    fn free_balance_of(who: [u8; 32]) -> u32;
+    fn reserved_balance_of(who: [u8; 32]) -> u32;
+}
+
+// #[extension(ExtensionFungibles)]
+// type Call;
+
+mod generated_by_extension_decl {
+
+    use super::*;
+    #[derive(Decode)]
+    pub enum ExtensionFungiblesCall<Impl: ExtensionFungibles> {
+        FreeBalanceOf { who: [u8; 32], _marker: PhantomData<Impl> },
```

Comment:
another way is to have another variant `__Unused(PhantomData<Impl>)` so that we don't need extra marker on other variants